lib/writer/index.js

```diff
@@ -226,6 +226,7 @@ class ImageWriter extends EventEmitter {
     progressStream.on('progress', (state) => {
       state.device = options.path
       state.type = 'write'
+      state.speed = target.speed
```

Comment:
Hmmm, so is this simply 'overwriting' the speed that we used to be setting from the ReadStream (or something) ?  Does it make sense to still be recording the ReadStream speed if we're now not using it?

Does a similar change to what you've done here need to be made in `_createVerifyPipeline` ?

Reply:
...or perhaps it'd make sense for _all_ the streams you added with pipage to have their own speedometer, to make it easier in the code elsewhere to pick'n'choose which speed(s) to display? \*shrug\*

Reply:
> used to be setting from the ReadStream (or something)

It's the progress-stream

> Does it make sense to still be recording the ReadStream speed if we're now not using it?

Yes, otherwise the progress-stream will not be able to calculate the ETA and other values, which are based on the source

> ...or perhaps it'd make sense for all the streams you added with pipage to have their own speedometer, to make it easier in the code elsewhere to pick'n'choose which speed(s) to display?

No, this is perf sensitive code, and speed metering does come with additional cost. Also the speed won't differ within the pipeline. The only reason the source & target speeds differ is when the source is compressed, and less data is read than is written.

>  Does a similar change to what you've done here need to be made in _createVerifyPipeline ?

No. The verify pipeline doesn't write to anywhere.

Reply:
Thanks for the clarification :+1:  Some of the answers now seem obvious, post-hangover ;-)

Although if the performance is really that sensitive, I guess the speedometer on the original progress-stream can be removed?

Reply:
Ah, nope, I've just found https://github.com/resin-io/etcher/blob/master/lib/writer/progress-stream.js#L76

I didn't realise the ETA was calculated like this by the progress-stream. (with the writer / writer-proxy / etc. and now with added pipage, it's easy to get confused about what part of the code is doing what :wink: )